modules/lemmaBidAdapter.js

```diff
@@ -0,0 +1,406 @@
+  return a.hostname;
+}
+
+function _getSiteObject(request, conf) {
+  var params = request && request.params ? request.params : null;
+  if (cfn(params) != '') {
+    var pubId = cfn(params.pubId) != '' ? params.pubId : '0';
+    var siteId = cfn(params.siteId) != '' ? params.siteId : '0';
+    var appParams = cfn(params.app);
+    if (!appParams) {
+      return {
+        publisher: {
+          id: pubId.toString()
+        },
+        domain: _getDomain(conf.pageURL),
+        id: siteId.toString(),
+        ref: conf.refURL,
+        page: conf.pageURL
+      };
+    }
+  }
+  return null;
+}
+
+function _getAppObject(request) {
+  var params = request && request.params ? request.params : null;
+  if (cfn(params) != '') {
+    var pubId = cfn(params.pubId) != '' ? cfn(params.pubId) : '0';
+    var appParams = cfn(params.app);
+    if (cfn(appParams)) {
+      return {
+        publisher: {
+          id: pubId.toString(),
+        },
+        id: appParams.id,
+        name: appParams.name,
+        bundle: appParams.bundle,
+        storeurl: appParams.storeUrl,
+        domain: appParams.domain,
+        cat: appParams.categories,
+        pagecat: appParams.page_category
+      };
+    }
+  }
+  return null;
+}
+
+function _getDeviceObject(request) {
+  var params = request && request.params ? request.params : null;
+  if (cfn(params) != '') {
+    return {
+      dnt: utils.getDNT() ? 1 : 0,
+      ua: navigator.userAgent,
+      language: (navigator.language || navigator.browserLanguage || navigator.userLanguage || navigator.systemLanguage),
+      w: (window.screen.width || window.innerWidth),
+      h: (window.screen.height || window.innerHeigh),
+      geo: {
+        country: params.country,
+        lat: params.latitude,
+        lon: params.longitude,
+        region: params.region,
+        city: params.city,
+        zip: params.zip
+      },
+      ip: params.ip,
+      devicetype: params.device_type,
+      ifa: params.ifa,
+    };
+  }
+  return null;
+}
+
+function setOtherParams(request, ortbRequest) {
+  var params = request && request.params ? request.params : null;
+  if (request && request.gdprConsent) {
+    ortbRequest.regs = { ext: { gdpr: request.gdprConsent.gdprApplies ? 1 : 0 } };
+    ortbRequest.user = { ext: { consent: request.gdprConsent.consentString } };
+  }
+  if (cfn(params) != '') {
+    ortbRequest.tmax = params.tmax;
+    ortbRequest.bcat = params.bcat;
+  }
+}
+
+function _getSizes(request) {
+  var size = cfn(cfn(request).sizes)[0];
+  if (utils.isArray(size) && cfn(size).length > 0) {
+    return size;
+  }
+  return null;
+}
+
+function _getBannerRequest(bid) {
+  var bObj;
+  var adFormat = [];
+  if (cfn(bid.mediaType) === 'banner' || utils.deepAccess(bid, 'mediaTypes.banner')) {
+    var params = cfn(bid) != '' ? bid.params : null;
+    var bannerData = cfn(params.banner);
+    var sizes = _getSizes(bid);
+    if (cfn(sizes).length == 0) {
+      sizes = cfn(bid.mediaTypes.banner.sizes[0]);
```

Comment:
Is this condition reachable? It looks like `_getSizes` will return null if length is 0.

Reply:
If the request object does not contain sizes parameter, then this condition reached. I have updated condition.